test/e2e/network_segmentation.go

```diff
@@ -612,8 +612,11 @@ var _ = Describe("Network Segmentation", func() {
 					By("delete pods in test namespace to unblock CUDN CR & associate NAD deletion")
 					_, err := e2ekubectl.RunKubectl(c.namespace, "delete", "pod", "--all")
 					Expect(err).NotTo(HaveOccurred())
-					_, err = e2ekubectl.RunKubectl("", "delete", "clusteruserdefinednetwork", c.name)
-					Expect(err).NotTo(HaveOccurred())
+					_, _ = e2ekubectl.RunKubectl("", "delete", "clusteruserdefinednetwork", c.name)
+					Eventually(func() error {
+						_, err := e2ekubectl.RunKubectl("", "get", "clusteruserdefinednetwork", c.name)
```

Comment:
hmm is it taking that long for the CUDNs to be deleted?
maybe `--wait` is better here? which blocks till deletion is complete? you can also use the --timeout=120s parameter with it

Reply:
It shouldn't take long, but since we dont wait on pod deletion and CUDN blocks deletion until no consuming pod exist anyways, we have this long timeout because pods are involved.

I changes it to specify --wait --timeout=120s instead of the eventuality and pod deletion to be done using go-client, PTAL